### PR TITLE
Faster tests when running locally.

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**
We make here efficient use of the docker cache to avoid having to download every time tensorflow. The tests also run in an isolated environement, which avoids polluting the project directory.

The command that google's CI runs for gpu tests is `/bin/bash tools/run_docker.sh -s linux -d gpu -c 'make gpu-unit-test'`.

Can we change that to `/bin/bash tools/run_google_cloud_tests.sh`? So that it's possible to modify it later on with pull requests?

### CODE REVIEW

File: run_google_cloud_tests.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+set -e
+
+bash tools/run_gpu_tests.sh
```
Comment (seanpmorgan) [inline run_google_cloud_tests.sh]: Similar to a little while ago, this requires Googlers to make a quick change on the GPU test script. Marking blocked until that is complete

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -56,9 +56,8 @@ Try these useful commands below:
 
 * Format code automatically: `bash tools/run_docker.sh -c 'make code-format'`
```
Comment (seanpmorgan) [inline CONTRIBUTING.md]: The makefile is now removed so this needs to be updated.

File: run_google_cloud_tests.sh (hunk 1 above)
Comment (gabrieldemarmiesse) [reply run_google_cloud_tests.sh]: Thanks! Hopefully it'll be general enough this time :)

File: CONTRIBUTING.md (hunk 2 above)
Comment (gabrieldemarmiesse) [reply CONTRIBUTING.md]: It's done in #1028 and I don't want to have conflicts :(

**Comment by gabrieldemarmiesse** [comment]
I don't know why I removed the makefile. Le me bring it back to life.

File: tools/run_docker.sh [hunk 3]
```diff
@@ -1,86 +0,0 @@
...
-fi
-
-DOCKER_CMD="${COMMAND}"
-echo "Docker image: ${DOCKER_IMAGE}"
-echo "Docker command: ${DOCKER_CMD}"
-docker run ${DOCKER_OPTS}                   \
-    --network=host                          \
-    --rm -v ${ROOT_DIR}:/addons -w /addons  \
-    ${DOCKER_IMAGE}                         \
-    /bin/bash -c "${DOCKER_CMD}"
```
Comment (seanpmorgan) [inline tools/run_docker.sh]: Please keep this file while we switch GPU CI.

Comment (gabrieldemarmiesse) [reply tools/run_docker.sh]: Sure thing.

Comment (gabrieldemarmiesse) [reply tools/run_docker.sh]: Done

**Pull request merged**